### ISSUE TITLE
wp_stream_query function backwards compatibility broken

### ISSUE THREAD

**Issue by sc0ttkclark** [xwp/stream#770]
The wp_stream_query function no longer exists in 3.0, which breaks code for plugins and themes using it from 2.x.

It's also still referenced here: https://github.com/xwp/stream/wiki/Query-Reference

**Comment by sc0ttkclark** [comment]
Looks like the way to query is now:

`$results = wp_stream_get_instance()->db->query->query( $args );`

A lot less clean looking for sure.

**Comment by lukecarbis** [comment]
Thanks for the report @sc0ttkclark. Yes, backwards compatibility with 2.0 is broken in this version. You're right that it's a lot less clean looking.

Starting in 3.0.2, I've added a helper function which cleans it up a little. You'll be able to do:

```
$results = wp_stream_get_instance()->db->query( $args );
```

Or my personal preference:

```
$stream = wp_stream_get_instance();
$stream->db->query( $args );
```

**Comment by sc0ttkclark** [comment]
You could still have the function around though and throw a `_doing_it_wrong()` -- https://developer.wordpress.org/reference/functions/_doing_it_wrong/

Why break it to break it?

**Comment by lukecarbis** [comment]
There's lots of breaking changes between version 2 and 3. Our policy is to maintain back compat between minor version releases, but not between major version releases, in the interest of keeping the code base as slim as possible.

**Comment by sc0ttkclark** [comment]
Gotcha, wasn't aware that was your plan. Thanks for following up with a quicker path to querying though, appreciate it.

**Comment by GaryJones** [comment]
@sc0ttkclark Non-sarcastic question - doesn't the major version bump between v2 and v3 suggest there's a strong chance of non-backwards-compatibility, as per [semantic versioning](http://semver.org)?

**Comment by sc0ttkclark** [comment]
Yes, based on Semantic versioning, that's what you'd expect. But the API didn't change (the args stayed the same) the way to call the query method changed. So now there's no wrapping function, which is fine, I would have preferred to use a forward-compatible solution in the code I built originally but the function was all that was in the examples for that.

However, in the WordPress / plugin world, semantic versioning is often too cut and dry, and backwards compatibility (even if it's enabled through a constant or hook) can be useful to ease in the transition. I thought Stream did a great job of the migration process to 3.0, there was just no code guide I had looked at and the examples were still using the old code.
